src/ocean/THCM.C

```diff
@@ -219,46 +207,27 @@ THCM::THCM(const Teuchos::ParameterList& params, Teuchos::RCP<Epetra_Comm> comm)
                 << xdist, __FILE__, __LINE__);
     }
 
-    hdim           = paramList.get("Depth hdim", 4000.0);
-    double qz      = paramList.get("Grid Stretching qz", 1.0);
-    int    itopo   = paramList.get("Topography", 1);
-    bool   flat    = paramList.get("Flat Bottom", false);
-    bool   rd_mask = paramList.get("Read Land Mask", false); //== false in experiment0
+    double qz      = paramList.get<double>("Grid Stretching qz");
```

Comment:
These and a couple of others are only used in the constructor to initialise parts of the Fortran code and/or initialise files (that are presumably read by the Fortran code). I wasn't sure if it was safe/sensible to allow setting them after the initial constructor creation?

If yes, then it's an easy fix. If no, then things get a bit messier/uglier...

Reply:
Like I said, anything related to the grid, which this is I think, should only be used in the initialization. If you want to change them, you should reinitialize everything.

Reply:
You mean this [Fortran initialisation](https://github.com/nlesc-smcm/i-emic/blob/d347c4d72e0a9d82a965b38e8de2a93ea476a2c2/src/ocean/THCM.C#L341-L348), right? Is it that initialise function expensive to run? (i.e. would it be ok to run everytime `setParameters` is called?) And is it fine to just call again?

Reply:
No, that is not doable. For instance during a continuation, you constantly want to adjust parameters, and every time you run the initialization, a new ocean will be created that is initialized to 0.

Reply:
Ok, then I think it'd actually be best to make a distinction between "default parameters for initialisation" and "other default parameters" (i.e. that can be tweaked at runtime), so we can distinguish the two and validate them properly.